types/src/transaction/transaction.rs

```diff
@@ -486,6 +632,38 @@ impl Decodable for Transaction {
                     nonce: d.val_at(5)?,
                 })
             }
+            ASSET_COMPOSE_ID => {
+                if d.item_count()? != 11 {
+                    return Err(DecoderError::RlpIncorrectListLen)
+                }
+                Ok(Transaction::AssetCompose {
+                    network_id: d.val_at(1)?,
+                    shard_id: d.val_at(2)?,
+                    world_id: d.val_at(3)?,
+                    metadata: d.val_at(4)?,
+                    registrar: d.val_at(5)?,
+                    inputs: d.list_at(6)?,
+                    output: AssetMintOutput {
+                        lock_script_hash: d.val_at(7)?,
+                        parameters: d.val_at(8)?,
+                        amount: d.val_at(9)?,
+                    },
+                    nonce: d.val_at(10)?,
+                })
+            }
+            ASSET_DECOMPOSE_ID => {
+                if d.item_count()? != 6 {
```

Comment:
7

Reply:
And it seems that you need more tests for decomposing.

Reply:
I fixed it and added an RLP test case for decompose transaction.